### ISSUE TITLE
Release 8.0.0

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#2575]
- [x] stylelint release
- [x] stylelint-config-recommended update/release
- [x] stylelint-config-standard update/release
- [x] stylelint-demo update
- [x] stylelint.io update
- [x] tweet

We've a [milestone](https://github.com/stylelint/stylelint/milestone/23) for `8.0.0`. There's a handful of remaining issues.

The new issue on the radar is ensuring compatibility with [`postcss@6`](https://github.com/postcss/postcss/releases/tag/6.0.0). I'll create an issue for that now.

**Comment by hudochenkov** [comment]
What our plans for 8.0 development? Should we merge `v8` into `master` now? What to do if we need to release a patch for 7.12?

**Comment by jeddy3** [comment]
> What our plans for 8.0 development? Should we merge v8 into master now? What to do if we need to release a patch for 7.12?

I think we should continue to work in the `v8` branch until we're about to release `8.0.0`. No new features in `7.0.`, though.

I'm currently in the process of merging `master` into `v8`, so it's up to date. It's as big a task as I thought it would be. I'm down to about 30 failing tests. Mainly around the `empty-line` before rules - in `master` we added `--fix` support and in `v8` we rewrote shared line comment support.

I'm going to push on for a bit more but I might have to throw up what's there soon.

**Comment by jeddy3** [comment]
Ok, I've merged everything. Caching and autofixing were added to `master` since the last merge into `v8`, so it was a fairly drawn-out process. I've patched up everything (linting, tests, snapshots etc), except for the `*-empty-line-before` tests:

```shell
Test Suites: 5 failed, 267 passed, 272 total
Tests:       31 failed, 13045 passed, 13076 total
Snapshots:   7 passed, 7 total
Time:        85.169s
Ran all test suites.
```

The five suites are the 5 `*-empty-line-before` rules. It's likely the same problem across all the suites, so hopefully, it'll just be the case of patching one rules' tests and rolling that out across the others.

I'm out of time I'm afraid, though. I might be able to take another look tomorrow evening. However, @hudochenkov if you fancy taking a stab at those tests, please do as you're very familiar with those rules. I think branching off `v8` and creating a PR would be the best way forward.

**Comment by hudochenkov** [comment]
Thanks, @jeddy3! I'll look into these tests at sometime today.

**Comment by hudochenkov** [comment]
In `v8` it's so much easier to find which test is failing!

![image](https://user-images.githubusercontent.com/654597/27607520-cd48b90a-5b8c-11e7-9d39-66fa8325915b.png)

Thanks, @davidtheclark, for [making this improvement](https://github.com/stylelint/stylelint/pull/2262/files#diff-1f69085782be126b5bbc8cedfb05c700)!

**Comment by hudochenkov** [comment]
> The five suites are the 5 `*-empty-line-before` rules. It's likely the same problem across all the suites, so hopefully, it'll just be the case of patching one rules' tests and rolling that out across the others.

After updating tests that miss `fixed` property, I can see that they are all different problems. There are common issue, but mostly they are different.

I'm working on fixes for these rules.

**Comment by hudochenkov** [comment]
Why [this test](https://github.com/stylelint/stylelint/blob/36bc35cb162c33cc519c0703473c5e92d2eda09f/lib/rules/at-rule-empty-line-before/__tests__/index.js#L215-L217) pass for v7?!

**Comment by evilebottnawi** [comment]
@hudochenkov why not? :smile: We can remove incompatibility tests

**Comment by hudochenkov** [comment]
I should not pass, because `@media {}` has block. Well, at least `v8` get it right :)

**Comment by jeddy3** [comment]
@stylelint/core When do we think is a good time to release `8.0.0`? I'm going to be mostly off-line for a month from the 20th of July and @hudochenkov is busy with his [recent move](https://github.com/stylelint/stylelint/pull/2704#issuecomment-312745594) (enjoy the Netherlands btw! 😄). So, it's likely to be a skeleton crew until mid-August. Do we delay `8.0.0` until then, or do I do a release before the 20th. I'm happy to but I just wanted to pre-warn that won't be around to help address any issues that arise.

**Comment by ntwb** [comment]
I think the v8 branch has had a fair bit of _scope creep_ in the ~8 months of its life...

Per the above, I think we release v8 ASAP 😄 

Those of us who are around can triage issues, and if needed we can cut a new release (see #2705), and anything else can wait until late August IMHO.

**Comment by jeddy3** [comment]
> I think the v8 branch has had a fair bit of scope creep in the ~8 months of its life...

Yes, totally! As I mentioned in https://github.com/stylelint/stylelint/issues/2545#issuecomment-312503532, I think we can do a major release whenever a bug fix or feature demands it, rather than grouping issues into one prelonged release.

> Per the above, I think we release v8 ASAP 😄

Sounds good. I'll do a release once those PRs are in.

**Comment by hudochenkov** [comment]
The bigger release, the more likely there are problems :) Let's release often.

What about new ESLint and Prettier: https://github.com/stylelint/stylelint/pull/2673, https://github.com/stylelint/stylelint/pull/2706? Are they holding 8.0.0? @ntwb, @jeddy3 can you finish them before July 20th?

**Comment by jeddy3** [comment]
> Are they holding 8.0.0? @ntwb, @jeddy3 can you finish them before July 20th?

Yes, they're ready to go. Just need to run prettier on the command line once the other PRs are in.

**Comment by ntwb** [comment]
> It's just these three PRs that need approving, and then we're good to go I think: #2713, #2709, #2702

Approved #2713 and #2709 
Will review #2702 once existing requested changes are added

**Comment by evilebottnawi** [comment]
Need wait `postcss-less` with support postcss `v6` and we can release 8.0.0, right?

**Comment by jwilsson** [comment]
> Need wait postcss-less with support postcss v6 and we can release 8.0.0, right?

I don't know if it's necessary, we've been using `postcss@6` and `postcss-less@1` in `lesshint` for a couple of weeks and we haven't had any issues reported.

**Comment by evilebottnawi** [comment]
@jwilsson We will have an extra `postcss` v5 package, it is not good

**Comment by jeddy3** [comment]
> @jwilsson We will have an extra postcss v5 package, it is not good

It sounds like it _isn't required_ for the release though and is something that can be addressed afterwards. Is that a fair conclusion?

**Comment by jwilsson** [comment]
I just wanted to let everyone know that we haven't seen any issues between the two libraries.

I agree that two different `postcss` packages is bad but it's all up to you!

**Comment by ntwb** [comment]
I don't think it will be an issue, that said, I just created a PR bumping PostCSS to 6.x

https://github.com/shellscape/postcss-less/pull/83

If all is good with that PR would cutting a new release of postcss-less 2.0.0 be the go?

**Comment by ntwb** [comment]
> It sounds like it isn't required for the release though and is something that can be addressed afterwards. Is that a fair conclusion?

If we merge v8 into master and spend a couple of days testing I think would reassure us?

**Comment by jeddy3** [comment]
> If we merge v8 into master and spend a couple of days testing I think would reassure us?

SGTM. Shall I merge `v8` into `master` now?

**Comment by ntwb** [comment]
Why not, looks good to me 👍

**Comment by jeddy3** [comment]
Just the https://github.com/stylelint/stylelint/pull/2706 & https://github.com/stylelint/stylelint/pull/2743 PRs in need of a review, and then I think we're ready to give `master` a test.

After that, we can release! Hopefully later today or tomorrow :)

**Comment by ntwb** [comment]
> Just the #2706 & #2743 PRs in need of a review

Both thumbs up from me, I've reviewed, approved, and updated as needed ✅ 

> After that, we can release! Hopefully later today or tomorrow :)

Excited for this release 😄

**Comment by jeddy3** [comment]
All merged.

@stylelint/core When you get a moment, can you try out `master` and if things look good I'll do a release! (FYI, I've tried it against the codebase I'm currently working on and it was fine)

**Comment by hudochenkov** [comment]
On my current project `master` works fine either. I lint `*.scss` and use `stylelint-order@0.5.0` and `stylelint-scss@1.5.1`.

**Comment by ntwb** [comment]
Just sipping my first coffee of the day now, will test _all the things_ in the next couple of hours 😄

**Comment by jeddy3** [comment]
@ntwb Excellent. Let us know how you get on. Other than that, I think we're good to go as I believe what PR can be merged have been merged, and what dependencies can be updated have been updated.

**Comment by ntwb** [comment]
Just testing it now... Will reply back shortly...

**Comment by ntwb** [comment]
Go for it... LGTM

**Comment by ntwb** [comment]
@jeddy Looking at commit https://github.com/stylelint/stylelint-config-recommended/commit/347bdd2060535ea97c24f2455895cb631e6fa04a and `"stylelint": "^8.0.0"` has been used, should we now per the updated semver guidelines use `"stylelint": "~8.0.0"`, i.e. `~` tilde rather than `^` caret?

**Comment by jeddy3** [comment]
@ntwb I think it's fine. `~` is for users of the stylelint in their builds. I think `^` is still correct for sharable config authors.

**Comment by ntwb** [comment]
Cool, just checking for my own sanity 😏

**Comment by jeddy3** [comment]
🎉 Done! 🎉 

Congrats everyone! :)

**Comment by ntwb** [comment]
Woot, thanks for packaging it all up @jeddy3 and a huge thanks to everyone who contributed  🎉 💯

**Comment by hudochenkov** [comment]
Fantastic! Yay!
